src/test/java/spoon/test/template/testclasses/EnumAccessTemplate.java

```diff
@@ -0,0 +1,31 @@
+package spoon.test.template.testclasses;
+
+import spoon.reflect.reference.CtTypeReference;
+import spoon.template.ExtensionTemplate;
+import spoon.template.Local;
+import spoon.template.Parameter;
+
+public class EnumAccessTemplate extends ExtensionTemplate {
+
+	public void method() throws Throwable {
+		_AnEnum_._value_.name();
+	}
+	
+	@Parameter("_AnEnum_")
+	CtTypeReference<?> anEnum;
+	
+	@Parameter
+	Enum _value_;
+
+	@Local
+	public EnumAccessTemplate(Enum anEmum, CtTypeReference<?> anEnum) {
+		this._value_ = anEmum;
+		this.anEnum = anEnum; 

```

Comment:
rename anEmum to enumValue?

could this be created from "anEmum" directly (Reference().create(anEmum.getClass())?